db/db_impl.cc

```diff
@@ -218,6 +219,11 @@ DBImpl::DBImpl(const DBOptions& options, const std::string& dbname)
   immutable_db_options_.Dump(immutable_db_options_.info_log.get());
   mutable_db_options_.Dump(immutable_db_options_.info_log.get());
   DumpSupportInfo(immutable_db_options_.info_log.get());
+
+  // always open the DB with 0 here, which means if preserve_deletes_==true
+  // we won't drop any deletion markers until SetPreserveDeletesSequenceNumber()
+  // is called by client and this seqnum is advanced.
+  preserve_deletes_seqnum_.store(0);
```

Comment:
should we initialize it to the current seqnum? since in case of DB reopen, we can't know whether deletes before the current seqnum were preserved. Maybe a future feature is persist this across restarts.

Reply:
that's an option to consider but that would change some semantics. Right now we don't have good place to keep that data persisted inside DB internal structures like manifest, so we rely on user to keep track of those seqnums. That's why here we set it to 0, saying we won't process any deletes (if preserve_deletes == true) until user calls SetPreserve...() and informs us what's the current cutoff seqnum is.